lib/widgets/message_list.dart

```diff
@@ -710,35 +725,39 @@ class RecipientHeaderDate extends StatelessWidget {
           //   https://developer.mozilla.org/en-US/docs/Web/CSS/font-variant-caps#all-small-caps
           fontFeatures: const [FontFeature.enable('c2sc'), FontFeature.enable('smcp')],
         ).merge(weightVariableTextStyle(context)),
-        _kRecipientHeaderDateFormat.format(
-          DateTime.fromMillisecondsSinceEpoch(message.timestamp * 1000))));
+        formatHeaderDate(
+          zulipLocalizations,
+          DateTime.fromMillisecondsSinceEpoch(message.timestamp * 1000,
+            isUtc: false),
+          now: DateTime.now())));
   }
 }
 
-final _kRecipientHeaderDateColor = const HSLColor.fromAHSL(0.75, 0, 0, 0.15).toColor();
-
-final _kRecipientHeaderDateFormat = DateFormat('y-MM-dd', 'en_US'); // TODO(#278)
-
-/// A widget with the distinctive chevron-tailed shape in Zulip recipient headers.
-class RecipientHeaderChevronContainer extends StatelessWidget {
-  const RecipientHeaderChevronContainer(
-    {super.key, required this.color, required this.child});
+@visibleForTesting
+String formatHeaderDate(
+  ZulipLocalizations zulipLocalizations,
+  DateTime dateTime, {
+  required DateTime now,
+}) {
+  if (dateTime.year == now.year &&
+      dateTime.month == now.month &&
+      dateTime.day == now.day) {
+    return "Today";
+  }
 
-  final Color color;
-  final Widget child;
+  final yesterday = now
+    .copyWith(hour: 12, minute: 0, second: 0, millisecond: 0, microsecond: 0)
+    .add(const Duration(days: -1));
```

Comment:
I believe I've heard of a locale where this wouldn't work — IIRC in parts of Ethiopia, the days run from midnight to midnight but the hours are counted from noon to noon.

But I'm not certain how to write correct logic for that: partly because I'm not sure I remember the details right, partly because I'm not sure how other software handles it, including `DateTime.copyWith`, and partly because it's just complicated to think about. So it wouldn't make sense to try to handle that case without writing tests for it. Since we don't have a reasonable way to control the timezone (as commented on that existing test below), I think it's best to just ignore that case at present.